common.mk

```diff
@@ -114,46 +114,49 @@ generator_temp: $(SCALA_SOURCES) $(sim_files) $(SCALA_BUILDTOOL_DEPS) $(EXTRA_GE
 		--top-module $(MODEL_PACKAGE).$(MODEL) \
 		--legacy-configs $(CONFIG_PACKAGE):$(CONFIG) \
 		$(EXTRA_CHISEL_OPTIONS))
+	sed -i '1d' $(ANNO_FILE)
+	echo "[" >> t.json
+	echo "{" >> t.json
+	echo "  \"class\": \"sifive.enterprise.firrtl.ModuleHierarchyAnnotation\"," >> t.json
+	echo "  \"filename\": \"./mod-he.json\"" >> t.json
+	echo "}," >> t.json
+	mv $(ANNO_FILE) t2.json
+	cat t.json t2.json > $(ANNO_FILE)
+	rm t.json t2.json
 
 .PHONY: firrtl
 firrtl: $(FIRRTL_FILE)
 
 #########################################################################################
 # create verilog files rules and variables
 #########################################################################################
-REPL_SEQ_MEM = --infer-rw --repl-seq-mem -c:$(MODEL):-o:$(TOP_SMEMS_CONF)
-HARNESS_CONF_FLAGS = -thconf $(HARNESS_SMEMS_CONF)
-
-TOP_TARGETS = $(TOP_FILE) $(TOP_SMEMS_CONF) $(TOP_ANNO) $(TOP_FIR) $(sim_top_blackboxes)
-HARNESS_TARGETS = $(HARNESS_FILE) $(HARNESS_SMEMS_CONF) $(HARNESS_ANNO) $(HARNESS_FIR) $(sim_harness_blackboxes)
+CIRCT_TARGETS = $(VSRC_SMEMS_FILE) $(VSRC_MODH_JSON)
 
 # DOC include start: FirrtlCompiler
 # NOTE: These *_temp intermediate targets will get removed in favor of make 4.3 grouped targets (&: operator)
 .INTERMEDIATE: firrtl_temp
-$(TOP_TARGETS) $(HARNESS_TARGETS): firrtl_temp
+$(CIRCT_TARGETS): firrtl_temp
 	@echo "" > /dev/null
```

Comment:
Is the plan to remove the Top/Harness differentiation? I'm fine with this, the VLSI tools shouldn't need to be passed a verily file with only TOP anyways.

Reply:
I plan on having two `.f`s that say what modules reside underneath Top and Harness (where the harness `.f` will include modules that reside in the Top as well).

Reply:
@harrisonliew Any thoughts on this? Also does Hammer need the `.f` black-box manifest files?

Reply:
Hammer can't use the `.f` files directly; the content of that file would need to be appended into the list of input files. That's trivial if the `.f` files are kept.

Reply:
Ok, now there is a `*.all.f` and `*.top.f` file which has all modules or just top modules within it. I think the VLSI makefile needs to be updated, @harrisonliew do you see anything that you might need?

Reply:
Looks good. Yes, the VLSI Makefile does need to be updated, pending this request:

There are use cases in which you want to run some submodule through the VLSI flow (with the `VLSI_TOP` Make variable), but then simulate that module within the entire system. Right now this requires netlist hacking to substitute the Verilog that came out of the VLSI flow. Could you make your `dump-mods.py` script dump the following:
- `*.all.f`: all modules (this has Harness too, right? Not that useful in VLSI flow, but might as well have it anyway)
- `*.top.f`: ChipTop as the top module by default, but if some additional option is given to the script (e.g. `--vlsi-top`), then the top module is the one in `--vlsi-top`
- `*.???.f`: not sure what to name this file, but it should contain the remainder of the hierarchy above ChipTop or the module given by `--vlsi-top` option. This would be given to sim tools, since the netlist coming out of syn/par won't have any of these modules.

Reply:
Ok for clarity you want the following:

`*.top.f` - Everything underneath `--dut-top`
`*.tb.f` - Everything underneath the topmost level (normally the testharness) that doesn't include the modules under `*.top.f`

Do you need CY to change the recipe (i.e. pass in `VLSI_TOP` in a switch) or do you just plan on re-running the `dump-mods` script with `VLSI_TOP`?

Reply:
If "underneath" is inclusive, then yes, that is correct.

It would be easiest if `VLSI_TOP` was passed in with a switch or something. No point duplicating the recipe.

Reply:
Can you remind me why you can't just override the `TOP` variable when running these simulations?

Reply:
Hm, interesting point. If we don't require specification of `TOP` for the Top/Harness separation anymore, then a separate `VLSI_TOP` may not be necessary? After all a simulation tool doesn't care about what's harness/dut, it just needs all the files.

Reply:
I'll leave the API such that what you want to synth. is called `TOP`.